apps/explorer/src/components/Toggle.tsx

```diff
@@ -0,0 +1,166 @@
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { motion, Transition } from 'framer-motion'
+import { LucideIcon } from 'lucide-react'
+import { useCallback } from 'react'
+import styled, { css, useTheme } from 'styled-components'
+
+import { onEnterOrSpace } from '@/utils/inputs'
+
+interface ToggleProps {
+  toggled: boolean
+  onToggle: (value: boolean) => void
+  disabled?: boolean
+  ToggleIcons?: [LucideIcon, LucideIcon]
+  handleColors?: [string, string]
+  label?: string
+  className?: string
+}
+
+// TODO: Share component in the shared lib?
+const toggleWidth = 40
+
+const Toggle = ({ toggled, onToggle, className, disabled, ToggleIcons, handleColors, label }: ToggleProps) => {
+  const theme = useTheme()
+  const [ToggleIconRight, ToggleIconLeft] = ToggleIcons ?? [undefined, undefined]
+
+  const toggleBackgroundVariants = {
+    off: {
+      backgroundColor: theme.name === 'light' ? 'rgba(0, 0, 0, 0.15)' : theme.bg.background2
+    },
+    on: { backgroundColor: handleColors ? theme.bg.background2 : theme.global.accent }
+  }
+
+  const handleContainerVariants = {
+    off: { left: 0 },
+    on: { left: toggleWidth / 2 }
+  }
+
+  const handleVariants = {
+    off: { backgroundColor: handleColors?.[0] ?? 'white' },
+    on: { backgroundColor: handleColors?.[1] ?? 'white' }
+  }
+
+  const toggleState = toggled ? 'on' : 'off'
+
+  const transition: Transition = { duration: 0.2, type: 'tween' }
+
+  const handleSwitch = useCallback(() => {
+    if (!disabled) {
+      onToggle(!toggled)
+    }
+  }, [disabled, toggled, onToggle])
+
+  const getToggleIconColor = (isActive: boolean) => (isActive ? 'white' : theme.font.tertiary)
+
+  return (
+    <StyledToggle
+      onClick={handleSwitch}
+      onKeyDown={(e) => onEnterOrSpace(e, handleSwitch)}
+      className={className}
+      aria-label={label}
+      aria-checked={toggled}
+      role="checkbox"
+      tabIndex={0}
+      toggled={toggled}
+      variants={toggleBackgroundVariants}
+      animate={toggleState}
+      transition={transition}
+      disabled={disabled}
+    >
+      <ToggleHandleContainer variants={handleContainerVariants} animate={toggleState} transition={transition}>
+        <ToggleHandle variants={handleVariants} animate={toggleState} transition={transition} />
+      </ToggleHandleContainer>
+      {ToggleIconRight && ToggleIconLeft && (
+        <ToggleContent>
+          <ToggleIconContainer>
+            <ToggleIconRight color={getToggleIconColor(!toggled)} size={16} className="toggle-icon" strokeWidth={2} />
+          </ToggleIconContainer>
+          <ToggleIconContainer>
+            <ToggleIconLeft color={getToggleIconColor(toggled)} size={16} className="toggle-icon" strokeWidth={2} />
+          </ToggleIconContainer>
+        </ToggleContent>
+      )}
+    </StyledToggle>
+  )
+}
+
+export default Toggle
+
+export const StyledToggle = styled(motion.div)<Omit<ToggleProps, 'onToggle'>>`
```

Comment:
Because it wasn't useful for a simple div :)  I'll only pick `disabled, to be more precise and also remove the `toggled` prop that was useless.

PS: this component is a copy paste from the DW. This could be shared in the future..!